### PR TITLE
feat(evm): improve language around EVM deployments

### PR DESCRIPTION
**Pull request by MicaiahReid**
This PR:
 - merges the `deploy_contract_create2` and `deploy_contract` commands into one - both deployments types can be made from the command
 - adds the `AtomicProxyDeploymentFactory` contract to the EVM addon, which powers the ability for a user to contracts via a proxy contract in one transaction
 - adds a build process to automatically compile txtx contracts
 - adds transaction log parsing
 - at the kit level, fixes how we evaluate maps

Here is an example runbook that deploys an implementation contract, a proxy contract with the implementation set to the previously deployed contract, and initializes the implementation contract:
```hcl
addon "evm" {
    rpc_api_url = input.rpc_api_url
    chain_id = input.chain_id
}

signer "deployer" "evm::secret_key" {
    secret_key = input.deployer_secret_key
}

variable "impl_salt" {
	value = evm::bytes32("0x0000000000000000000000000000000000000000000000000000000000000001")
}
variable "proxy_salt" {
	value = evm::bytes32("0x0000000000000000000000000000000000000000000000000000000000000002")
}

action "deploy" "evm::deploy_contract" {
	contract = evm::get_contract_from_foundry_project("SimpleStorage")
	signer = signer.deployer

	create2 {
		salt = variable.impl_salt
	}
	proxy {
		create2 {
			salt = variable.proxy_salt
		}
	}
	initializer {
		function_name = "initialize"
		function_args = [42]
	}
}

variable "logs" {
	value = action.deploy.logs
}

output "impl_address" {
	value = action.deploy.impl_contract_address
}

output "proxy_address" {
	value = action.deploy.proxy_contract_address
}

output "contract_address" {
	value = action.deploy.contract_address
}

output "favorite_number" {
	value = jq(index(variable.logs, 4), ".data.favoriteNumber")
}
```

### CODE REVIEW

File: addons/evm/src/contracts/src/AtomicProxyDeploymentFactory.sol [hunk 1]
```diff
@@ -0,0 +1,79 @@
+// SPDX-License-Identifier: MIT
+pragma solidity ^0.8.20;
+pragma experimental ABIEncoderV2;
+
+import "@openzeppelin/contracts/proxy/ERC1967/ERC1967Proxy.sol";
```
Comment (lgalabru) [inline addons/evm/src/contracts/src/AtomicProxyDeploymentFactory.sol:5]: Hmm. I don't know how I feel about having gitsubmodules. What would be the alternatives?

Comment (MicaiahReid) [reply addons/evm/src/contracts/src/AtomicProxyDeploymentFactory.sol:5]: Yeah, I don't like it either. This is the standard way for managing contract dependencies with foundry.

We could have a separate repo for our evm contracts, but that introduces a whole CI process for cloning another repo when we make a build.

Is there any problem with just copying out the contract?

Comment (lgalabru) [reply addons/evm/src/contracts/src/AtomicProxyDeploymentFactory.sol:5]: Let's copy the contract for now and backlog this as something we need to revisit?

Comment (MicaiahReid) [reply addons/evm/src/contracts/src/AtomicProxyDeploymentFactory.sol]: Done! Task in planner [here](https://teams.microsoft.com/l/entity/com.microsoft.teamspace.tab.planner/mytasks?tenantId=0f1e1f86-ff36-4233-ba7c-e2d48a07bb41&webUrl=https%3A%2F%2Ftasks.teams.microsoft.com%2Fteamsui%2FpersonalApp%2Falltasklists&context=%7B%22subEntityId%22%3A%22%2Fv1%2Fpremiumplan%2Fc7a7d962-0836-4ed0-8a67-774535750432%2Forg%2Fdeb736b0-d91b-ef11-9f85-000d3a537223%2Fview%2Fboard%2Ftask%2FFB1E8DBC-10DA-4146-8580-175630BFBF04%22%7D)

**Review by lgalabru — APPROVED** [review]
Ship it 🚀

**Pull request merged**